classes/submission/PKPSubmissionMetadataFormImplementation.inc.php

```diff
@@ -147,6 +154,18 @@ function execute($submission, $request) {
 		if (empty($supportedSubmissionLocales)) $supportedSubmissionLocales = array($context->getPrimaryLocale());
 		if (in_array($newLocale, $supportedSubmissionLocales)) $submission->setLocale($newLocale);
 
+		// Versioning
+		if ($request->getUserVar('submissionRevision')) {
+			$revision = (int)$request->getUserVar('submissionRevision');
+		} else {
+			$context = $request->getContext();
+			$contextId = $context->getId();
+			$revision = $submission->getCurrentVersionId($contextId);
```

Comment:
submission function do not need context parameter, because every submission has it already